### PR TITLE
Fixes #35349 - deploy a foreman-release.deb symlink

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: puppet/modules/freight/templates/deploy_debs.erb [hunk 1]
```diff
@@ -43,6 +47,17 @@ files.each do |file|
 end
 system "/usr/bin/freight-cache -v -c /home/freight/freight.conf apt/#{os}"
 
+foreman_release_dir = "#{ENV['TARGETCACHE']}/pool/#{os}/#{version}/f/foreman-release/"
+release_files = Dir.glob("#{foreman_release_dir}/foreman-release_*.deb")
+# we assume the file names can be alphabetically sorted to find the newest
```
Comment (ekohl) [inline puppet/modules/freight/templates/deploy_debs.erb:52]: That is assuming we never hit a version 10, right?

File: puppet/modules/freight/templates/deploy_debs.erb [hunk 2]
```diff
@@ -43,6 +47,17 @@ files.each do |file|
 end
 system "/usr/bin/freight-cache -v -c /home/freight/freight.conf apt/#{os}"
 
+foreman_release_dir = "#{ENV['TARGETCACHE']}/pool/#{os}/#{version}/f/foreman-release/"
```
Comment (ekohl) [inline puppet/modules/freight/templates/deploy_debs.erb:50]: I'm tempted to store the path relative to the target cache in a separate var and reuse it in `fastly-purge`.

**Review by ekohl — APPROVED** [review]

File: puppet/modules/freight/templates/deploy_debs.erb [hunk 3]
```diff
@@ -43,6 +47,17 @@ files.each do |file|
...
+release_files = Dir.glob("#{foreman_release_dir}/foreman-release_*.deb")
+# we assume the file names can be alphabetically sorted to find the newest
+latest_release_file = release_files.sort.last
```
Comment (evgeni) [inline puppet/modules/freight/templates/deploy_debs.erb:53]: Well, this assumption is clearly wrong, I am stupid, and need to think more about that!

```
> ["foreman-release_3.4.0-1_all.deb", "foreman-release_3.4.0~rc1-1_all.deb", "foreman-release_3.4.0~rc2-1_all.deb"].sort.last
=> "foreman-release_3.4.0~rc2-1_all.deb"
```

File: puppet/modules/freight/templates/deploy_debs.erb (hunk 2 above)
Comment (evgeni) [reply puppet/modules/freight/templates/deploy_debs.erb:50]: I was thinking about that, yeah!

File: puppet/modules/freight/templates/deploy_debs.erb (hunk 1 above)
Comment (evgeni) [reply puppet/modules/freight/templates/deploy_debs.erb:52]: debian revision 10? yeah, that too, but also https://github.com/theforeman/foreman-infra/pull/1768#discussion_r943273530

File: puppet/modules/freight/templates/deploy_debs.erb (hunk 3 above)
Comment (evgeni) [reply puppet/modules/freight/templates/deploy_debs.erb]: mtime it is!

File: puppet/modules/freight/templates/deploy_debs.erb (hunk 1 above)
Comment (evgeni) [reply puppet/modules/freight/templates/deploy_debs.erb]: now sorting by mtime, this should be safe enough.

File: puppet/modules/freight/templates/deploy_debs.erb [hunk 4]
```diff
@@ -43,6 +47,18 @@ files.each do |file|
 end
 system "/usr/bin/freight-cache -v -c /home/freight/freight.conf apt/#{os}"
 
+foreman_release_pool = "pool/#{os}/#{version}/f/foreman-release"
+foreman_release_dir = "#{ENV['TARGETCACHE']}/#{foreman_release_pool}"
+release_files = Dir.glob("#{foreman_release_dir}/foreman-release_*.deb")
+release_files.sort! {|x,y| File.mtime(x) <=> File.mtime(y)}
+latest_release_file = release_files.last
```
Comment (ekohl) [inline puppet/modules/freight/templates/deploy_debs.erb:54]:
```suggestion
latest_release_file = release_files.max_by { |f| File.mtime(f) }
```

Comment (evgeni) [reply puppet/modules/freight/templates/deploy_debs.erb]: TIL!

**Review by ekohl — APPROVED** [review]

File: puppet/modules/freight/templates/deploy_debs.erb (hunk 4 above)
Comment (ekohl) [reply puppet/modules/freight/templates/deploy_debs.erb]: https://ruby-doc.org/core-2.7.1/Enumerable.html#method-i-max_by is very nice

**Pull request merged**